src/cpp/src/tokenizer/make_tokenizer_stateful.cpp

```diff
@@ -165,18 +185,31 @@ bool ov::genai::MakePaddingSatateful::run_on_model(const std::shared_ptr<ov::Mod
     // Save targets before adding new target with ReadValue to avoid recursion.
     auto target_inputs = const_node->output(0).get_target_inputs();
     auto max_length_rv = std::make_shared<v6::ReadValue>(default_max_length_const, max_length_var);
-    auto subtract_node = std::make_shared<v1::Subtract>(max_length_rv, num_added_tokens_const);
+    model->add_sinks({std::make_shared<v6::Assign>(max_length_rv, max_length_var)});
+    model->add_variables({max_length_var});
+    auto max_length_node = std::make_shared<v1::Subtract>(max_length_rv, num_added_tokens_const);
+
+    var_info = {ov::Shape{}, ov::element::boolean, ov::genai::IS_MAX_LENGTH_SET};
+    auto is_max_len_set_var = std::make_shared<op::util::Variable>(var_info);
+    auto defaul_false_const = std::make_shared<v0::Constant>(ov::element::boolean, ov::Shape{}, std::vector{false});
+    auto is_max_len_set_rv = std::make_shared<v6::ReadValue>(defaul_false_const, is_max_len_set_var);
+    auto is_max_len_set_assign = std::make_shared<v6::Assign>(is_max_len_set_rv, is_max_len_set_var);
+    model->add_sinks({is_max_len_set_assign});
+
+    auto int32_max_constant = std::make_shared<v0::Constant>(ov::element::i32, ov::Shape{}, std::vector{std::numeric_limits<int32_t>::max() - 1000});
+    // auto int32_max_constant = std::make_shared<v0::Constant>(ov::element::i32, ov::Shape{}, std::vector{10000});
```

Comment:
```suggestion
```

Reply:
removed. There is a strange behavior regarding int32_max left a TODO.